src/gpt_resolve/pdf_generator.py

```diff
@@ -59,10 +83,22 @@ def generate_solutions_pdf(
         solutions_dir.glob("*_solution.txt"),
         key=lambda x: int(x.stem.split("_")[0][1:]),
     )
+        # Get all solution files sorted
+    solution_files = sorted(
+        solutions_dir.glob("*_solution.txt"),
+        key=lambda x: int(x.stem.split("_")[0][1:])
+    )
 
+    sol_files_with_issues = []
     # Add each solution to document
     for sol_file in solution_files:
+        print(f"Validating solution: {sol_file.name}")
         content = sol_file.read_text(encoding="utf-8")
+
+        if not validate_latex_content(content):
+            print(f"Error: Solution '{sol_file.name}' has LaTeX syntax errors. Please fix it.")
+            sol_files_with_issues.append(sol_file)
+            continue
```

Comment:
Nice! The only thing left here is that any errors will count in sol_files_with_issues and loop will continue, but in the end the code will try to compile the whole pdf again, generating the same error that currently happens (for many questions, the traceback will be big).

So if you print errors only outside the loop in case of sol_files_with_issues not empty, it'd be solved:
```
for sol_file in solution_files:
    ...
    if not validate_latex_content(content):
        sol_files_with_issues.append(sol_file)
        continue

if sol_files_with_issues:
    print("Errors in one or more solutions:\n' + '\n'.join(sol_files_with_issues) + '\n\nPlease check for syntax errors")
...
```